functions.opengraph.php

```diff
@@ -293,17 +293,29 @@ function jetpack_og_get_image( $width = 200, $height = 200, $max_images = 4 ) {
 	if ( empty( $image ) && function_exists( 'blavatar_domain' ) ) {
 		$blavatar_domain = blavatar_domain( site_url() );
 		if ( blavatar_exists( $blavatar_domain ) ) {
-			$image_url = blavatar_url( $blavatar_domain, 'img', $width, false, true );
-
 			$img_width  = '';
 			$img_height = '';
-			$cached_image_id = get_transient( 'jp_' . $image_url );
-			if ( false !== $cached_image_id ) {
-				$image_id = $cached_image_id;
-			} else {
+
+			$image_url = blavatar_url( $blavatar_domain, 'img', $width, false, true );
+
+			// Build a hash of the Image URL. We'll use it later when building the transient.
+			if ( $image_url ) {
+				/**
+				 * Transient names are 45 chars max.
+				 * Let's generate a hash that's never more than 40 chars long.
+				 */
+				$image_hash = sha1( $image_url );
+			}
+
+			// Look for data in our transient. If nothing, let's get an attachment ID.
+			$cached_image_id = get_transient( 'jp_' . $image_hash );
+			if ( ! is_int( $cached_image_id ) ) {
```

Comment:
Honestly, I'm unsure.  If it works now, good enough, we can reevaluate later or third-parties can cast to int before saving.

I just wasn't sure if the number would get cast to a string when saved in the db.  Just being cautious.